go/buildhelper/helpertypes/types.go

```diff
@@ -0,0 +1,130 @@
+package helpertypes
+
+import (
+	"bytes"
+	"compress/gzip"
+	b64 "encoding/base64"
+	"io/ioutil"
+	"strings"
+
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/obscuronet/obscuro-playground/go/buildhelper/buildconstants"
+	"github.com/obscuronet/obscuro-playground/go/obscurocommon"
+)
+
+var IsRealEth bool
+
+func UnpackL1Tx(tx *types.Transaction) *obscurocommon.L1TxData {
+	if !IsRealEth {
+		t := obscurocommon.TxData(tx)
+		return &t
+	}
+	// ignore transactions that are not calling the contract
+	if tx.To() == nil || tx.To().Hex() != buildconstants.CONTRACT_ADDRESS.Hex() || len(tx.Data()) == 0 {
+		return nil
+	}
+
+	contractABI, err := abi.JSON(strings.NewReader(buildconstants.CONTRACT_ABI))
+	if err != nil {
+		panic(err)
+	}
+
+	//log.Log(fmt.Sprintf("Unpacking data: %b", tx.Data()))
+	method, err := contractABI.MethodById(tx.Data()[:4])
```

Comment:
@otherview I think the `4` here should be a constant (e.g. `methodBytesLen` or something) - hard to understand what's being done otherwise.

Reply:
@otherview It looks like the slicing is done by the `MethodById` method, so you don't need to slice it yourself.

Reply:
Good point 👍